pallets/automation-time/src/lib.rs

```diff
@@ -765,33 +797,73 @@ pub mod pallet {
 			<T as Config>::WeightInfo::run_native_transfer_task()
 		}
 
+		/// Determines if a task has been completed.
+		/// 
+		/// A task has been completed if all execution times exist in the past and/or are in the current time slot.
+		/// This means there a no more execution times to be run.
+		fn is_completed_task(execution_times: Vec<UnixTime>) -> bool {
+			if execution_times.len() == 0 { return true };
+
+			let current_time_slot = match Self::get_current_time_slot() {
+				Ok(time_slot) => time_slot,
+				// This will only occur for the first block in the chain.
+				Err(_) => return false,
+			};
+
+			let scheduled_times = Self::clean_execution_times_vector(execution_times);
+			let is_complete = *scheduled_times.last().unwrap() <= current_time_slot;
+			return is_complete;
+		}
+
+		/// Removes the task of the provided task_id and all scheduled tasks, including those in the task queue.
+		///
 		fn remove_task(task_id: T::Hash, task: Task<T>) {
 			let mut found_task: bool = false;
-			match Self::get_scheduled_tasks(task.time) {
-				None => {
-					let mut task_queue = Self::get_task_queue();
-					for i in 0..task_queue.len() {
-						if task_queue[i] == task_id {
-							task_queue.remove(i);
-							TaskQueue::<T>::put(task_queue);
-							found_task = true;
-							break
-						}
+			let execution_times = Self::clean_execution_times_vector(task.execution_times.to_vec());
+			let current_time_slot = match Self::get_current_time_slot() {
+				Ok(time_slot) => time_slot,
+				// This will only occur for the first block in the chain.
+				Err(_) => 0,
```

Comment:
@rhuttman can you double check my thinking here?  if `get_current_time_slot` does not exist then we want to guarantee this block gets hit - https://github.com/OAK-Foundation/OAK-blockchain/pull/128/files#diff-e6a13f7032a383e19b77d20043ca46041d94f5e49396812e2de1b2000c301e59R849-R866

Reply:
that will work